client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -29,10 +31,25 @@ const DomainRegistrationSuggestion = React.createClass( {
 			isAdded = hasDomainInCart( this.props.cart, suggestion.domain_name ),
 			buttonClasses,
 			buttonContent,
-			domainFlag = null;
+			domainFlags = [],
+			translate = this.props.translate;
 
 		if ( suggestion.domain_name ) {
-			domainFlag = <DomainSuggestionFlag domain={ suggestion.domain_name }/>;
+			const newTLDs = [];
+
+			if ( newTLDs.some(
+					( tld ) => endsWith( suggestion.domain_name, tld ) && suggestion.domain_name.substring( 0, suggestion.domain_name.length - ( tld.length + 1 ) ).indexOf( '.' ) === -1
+				) ) {
+				domainFlags.push( <DomainSuggestionFlag content={ translate( 'New', { context: 'Domain suggestion flag' } ) }/> );
+			}
+		}
+
+		if ( suggestion.isRecommended ) {
+			domainFlags.push( <DomainSuggestionFlag content={ translate( 'Recommended', { context: 'Domain suggestion flag' } ) }/> );
```

Comment:
There's no definitive answer :)

In some cases, a context could be useful, because `Recommended` or even `New` can be translated differently depending on the object (i.e New Post or New Page).

What is "Recommended" or "New"? A domain name? then the context should be just "domain name". But if the label is not that specific, and it refers to something a bit abstract ("a suggestion"), I'd just drop the context (for all the flags).